Packs/SophosCentral/Integrations/SophosCentral/SophosCentral.py

```diff
@@ -0,0 +1,1813 @@
+    alert_id = args.get('alert_id', '')
+    try:
+        result = client.get_alert(alert_id)
+        table_headers = ['id', 'description', 'severity', 'raisedAt', 'allowedActions',
+                         'managedAgentId', 'category', 'type']
+        object_data = create_alert_output(result, table_headers)
+        readable_output = tableToMarkdown(name='Found Alert:', t=object_data, headers=table_headers,
+                                          removeNull=True)
+        return CommandResults(outputs_key_field='id', outputs_prefix='SophosCentral.Alert',
+                              raw_response=result, outputs=object_data,
+                              readable_output=readable_output)
+    except DemistoException:
+        return CommandResults(readable_output=f'Unable to find the following alert: {alert_id}')
+
+
+def sophos_central_alert_action_command(client: Client, args: dict) -> CommandResults:
+    """
+    Take an action against a specific alert or alerts.
+
+    Args:
+        client (Client): Sophos Central API client.
+        args (dict): All command arguments
+
+    Returns:
+        CommandResults: outputs, readable outputs and raw response for XSOAR.
+    """
+    outputs = []
+    table_headers = ['id', 'action', 'completedAt', 'alertId', 'result',
+                     'requestedAt', 'startedAt', 'status']
+    alert_ids = argToList(args.get('alert_id'))
+    failed_alerts = []
+    results = []
+    for alert_id in alert_ids:
+        try:
+            result = client.action_alert(alert_id, args.get('action', ''), args.get('message', ''))
+            results.append(result)
+            object_data = {field: result.get(field) for field in table_headers}
+            outputs.append(object_data)
+        except DemistoException:
+            failed_alerts.append(alert_id)
+    readable_output = tableToMarkdown(name='Alerts Acted Against:', t=outputs,
+                                      headers=table_headers, removeNull=True)
+    if failed_alerts:
+        readable_output += f'\nAlerts not acted against: {",".join(failed_alerts)}'
+    return CommandResults(outputs_key_field='id', outputs_prefix='SophosCentral.AlertAction',
+                          raw_response=results, outputs=outputs,
+                          readable_output=readable_output)
+
+
+def sophos_central_alert_search_command(client: Client, args: dict) -> CommandResults:
+    """
+    Search for alerts based on query parameters.
+
+    Args:
+        client (Client): Sophos Central API client.
+        args (dict): All command arguments
+
+    Returns:
+        CommandResults: outputs, readable outputs and raw response for XSOAR.
+    """
+    start_date = args.get('start')
+    end_date = args.get('end')
+    date_range = args.get('date_range')
+    if date_range:
+        start_date, end_date = parse_date_range(date_range=date_range, date_format=DATE_FORMAT)
+        start_date = start_date[:-4] + 'Z'
+        end_date = end_date[:-4] + 'Z'

```

Comment:
why you are changing the string created in `parse_date_range`, and not your global `DATE_FORMAT` to fit to the format that you need?

Reply:
I haven't found a way to remove the microseconds but not the milliseconds in any other way, but if it is too destructive I can just use seconds. There shouldn't be a loss of incidents anyway.

Reply:
It is ok to do that in `list-alerts` cmd, but not in fetch incidents. so we'll have to think of a nicer solution anyhow. let's stay 5 min on the call on wednesday.